contracts/test/proposals/TestProposals.sol

```diff
@@ -0,0 +1,103 @@
+pragma solidity =0.8.13;
+
+import {console} from "hardhat/console.sol";
+
+import {Test} from "../../../forge-std/src/Test.sol";
+import {Addresses} from "./Addresses.sol";
+import {Proposal} from "./proposalTypes/Proposal.sol";
+
+import {vip15} from "./vips/vip15.sol";
+import {vip16} from "./vips/vip16.sol";
+import {vip17} from "./vips/vip17.sol";
+
+/*
+How to use:
+forge test --fork-url $ETH_RPC_URL --match-contract TestProposals -vvv
+
+Or, from another Solidity file (for post-proposal integration testing):
+    TestProposals proposals = new TestProposals();
+    proposals.setUp();
+    proposals.setDebug(false); // don't console.log
+    proposals.testProposals();
+    Addresses addresses = proposals.addresses();
+*/
+
+contract TestProposals is Test {
+    Addresses public addresses;
+    Proposal[] public proposals;
+    bool public DEBUG = true;
```

Comment:
would be great if this could be loaded from an environment variable

Reply:
that's doable but then the script don't run if we don't define this environment variable

Reply:
we could just have a default value, that starts out as true, and then if it reads in a string from the environment that says false, flip the flag to false.